google/cloud/spanner/internal/connection_impl.cc

```diff
@@ -379,90 +416,139 @@ QueryResult ConnectionImpl::ExecuteQueryImpl(
   if (esp.partition_token) {
     request.set_partition_token(*std::move(esp.partition_token));
   }
+  request.set_query_mode(query_mode);
 
-  auto const& stub = stub_;
-  // Capture a copy of `stub` to ensure the `shared_ptr<>` remains valid through
-  // the lifetime of the lambda. Note that the local variable `stub` is a
-  // reference to avoid increasing refcounts twice, but the capture is by value.
-  auto factory = [stub, request](std::string const& resume_token) mutable {
-    request.set_resume_token(resume_token);
-    auto context = google::cloud::internal::make_unique<grpc::ClientContext>();
-    return google::cloud::internal::make_unique<DefaultPartialResultSetReader>(
-        std::move(context), stub->ExecuteStreamingSql(*context, request));
-  };
-  auto rpc = google::cloud::internal::make_unique<PartialResultSetResume>(
-      std::move(factory), Idempotency::kIdempotent, retry_policy_->clone(),
-      backoff_policy_->clone());
-  auto reader = PartialResultSetSource::Create(std::move(rpc));
-
+  auto reader = retry_resume_fn(request);
   if (!reader.ok()) {
-    return QueryResult(
-        google::cloud::internal::make_unique<StatusOnlyResultSetSource>(
-            std::move(reader).status()));
+    return std::move(reader).status();
   }
   if (s.has_begin()) {
     auto metadata = (*reader)->Metadata();
     if (!metadata || metadata->transaction().id().empty()) {
-      return QueryResult(
-          google::cloud::internal::make_unique<StatusOnlyResultSetSource>(
-              Status(StatusCode::kInternal,
-                     "Begin transaction requested but no transaction returned "
-                     "(in ExecuteQuery).")));
+      return T(google::cloud::internal::make_unique<StatusOnlyResultSetSource>(
```

Comment:
Ideally in the case of a DmlResult, this would just return the `Status`, not a `DmlResult` wrapping the Status. Is there anyway to do that now?

Reply:
Yes, good catch.